src/core/config/exceptions.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <stdexcept>
+
+namespace config {
+
+class ConfigurationError : public std::exception {
+    std::string message_;
+
+public:
+    explicit ConfigurationError(std::string message) : message_(std::move(message)) {}
+
+    const char* what() const noexcept override {
+        return message_.data();
+    }
+};
+
+class UnknownOption : public ConfigurationError {
+    using ConfigurationError::ConfigurationError;
+};
```

Comment:
I think part of an exception message could be derived from the exception type and we should use this.
For example, almost any exception of type `UnknownOption` can start its message with "Unknown option: " with option name concatenated after. Or, as it is implemented right now, `"Option named " option name here " does not exist"`.
So I propose to add a constructor(s) from params from which the error message will be constructed using predefined message specific for exception type. This will unify exception messages and reduce amount of boilerplate (look, for example, at `InvalidChoice` usage in `src/python_bindings/py_to_any.cpp`)